tests/testthat/test-translate-sql-conditional.R

```diff
@@ -153,3 +153,95 @@ test_that("conditionals check arguments", {
 
   expect_snapshot(error = TRUE, translate_sql(switch(x, 1L, 2L)))
 })
+
+
+# case_match --------------------------------------------------------------
+
+test_that("LHS can handle different types", {
+  expect_equal(
+    translate_sql(case_match(z, 1L ~ "a")),
+    sql("CASE WHEN (`z` IN (1)) THEN 'a' END")
+  )
+
+  expect_equal(
+    translate_sql(case_match(z, y ~ "a")),
+    sql("CASE WHEN (`z` IN (`y`)) THEN 'a' END")
+  )
+
+  expect_equal(
+    translate_sql(case_match(z, f(y) ~ "a")),
```

Comment:
Maybe user a real function here since we expect to make this a warning at some point in the future?